testing/src/test/java/io/stargate/it/cql/SystemTablesTest.java

```diff
@@ -1,29 +1,94 @@
 package io.stargate.it.cql;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
 
 import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.ResultSet;
 import com.datastax.oss.driver.api.core.cql.Row;
+import com.datastax.oss.driver.api.core.cql.SimpleStatement;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.google.common.collect.Streams;
 import io.stargate.it.BaseOsgiIntegrationTest;
 import io.stargate.it.driver.CqlSessionExtension;
 import io.stargate.it.storage.StargateConnectionInfo;
 import io.stargate.it.storage.StargateEnvironmentInfo;
-import java.net.UnknownHostException;
+import io.stargate.it.storage.StargateSpec;
+import java.net.InetAddress;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 
+@StargateSpec(nodes = 2)
 @ExtendWith(CqlSessionExtension.class)
-public class SystemTablesTest extends BaseOsgiIntegrationTest {
+class SystemTablesTest extends BaseOsgiIntegrationTest {
+  @Test
+  @DisplayName("Should expose Stargate addresses in system.local and system.peers")
+  public void querySystemLocalAndPeers(CqlSession session) {
+    Iterator<Node> nodes = session.getMetadata().getNodes().values().iterator();
+
+    Node localNode = nodes.next();
+    Row localRow =
+        session
+            .execute(
+                SimpleStatement.builder("SELECT * FROM system.local").setNode(localNode).build())
+            .one();
+    assertThat(localRow).isNotNull();
+    assertThat(localRow.getInetAddress("listen_address"))
+        .isEqualTo(localNode.getListenAddress().map(s -> s.getAddress()).orElse(null));
+
+    ResultSet rs =
+        session.execute(
+            SimpleStatement.builder("SELECT * FROM system.peers").setNode(localNode).build());
```

Comment:
I'd split these test into separate methods for better isolation... is there any reason to keep them together?

Reply:
They're a bound together because the test explicitly sets the queried node using `setNode(localNode)`. I expect that `localNode` will not appear in the the `peers` table.

Reply:
Yep, now that I see the importance of the `.setNode` call, it makes sense :facepalm: